pkg/controller/reconciler.go

```diff
@@ -69,22 +69,22 @@ func (r *reconciler) InjectAPIReader(reader client.Reader) error {
 
 func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	obj := r.typ.DeepCopyObject().(client.Object)
+	obj.SetName(request.Name)
+	obj.SetNamespace(request.Namespace)
+	logger := r.logger.WithValues("name", obj.GetName(), "namespace", obj.GetNamespace())
+
 	if err := r.client.Get(ctx, request.NamespacedName, obj); err != nil {
 		if apierrors.IsNotFound(err) {
+			if r.shouldEnsureDeleted {
+				return r.ensureDeleted(ctx, obj, logger)
```

Comment:
`ensureDeleted` sounded quite confusing to me at first. 
Maybe `actuatorDelete` or something similar?

Reply:
I think "ensure deleted" is rather accurate - the goal is to ensure that the deletion has been correctly reconciled for a non-existing object. If we call the deletion reconciliation "delete" (which is arguably already confusing, but is nevertheless part of our "jargon"), then calling this "ensure deleted" must make sense as well.

Reply:
Ok, fine